internal/controllers/extension_controller.go

```diff
@@ -123,22 +147,69 @@ func (r *ExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alpha1.Ext
 		return ctrl.Result{}, nil
 	}
 
-	// TODO: kapp-controller integration
-	// * gather variables for resolution
-	// * perform resolution
-	// * lookup the bundle in the solution/selection that corresponds to the Extension's desired Source
-	// * set the status of the Extension based on the respective deployed application status conditions.
+	if !r.HasKappApis {
```

Comment:
No, this is evaluated only once when the controller starts. This is intentional since we also start the informers for watching App resources when the controller starts. As of today, there is no way for controller-runtime to dynamically add watches for resource types (though I remember some work on stopping and starting controllers was on the way).

Reply:
It is possible to add watches for resource types after the fact, but I would call it overkill.

In fact, I was going to suggest that maybe we don't even need this check since we have a feature gate. If someone enables the feature gate without installing kapp-controller, that a somewhat obvious configuration issue.